edivorce/apps/core/static/js/ajax.js

```diff
@@ -5,6 +5,23 @@ var ajaxOnChange = function () {
     // show/hide additional information if needed
     reveal(el);
 
+    var question = el.prop('name');
+    var value = getValue(el, question);
+    var isValid = true;
+
+    // Check if date is in valid format DD/MM/YYYY
+    if (el.is(".date-picker") || el.is(".date-pickers") || el.is(".date-picker-future")){
+        isValid = validateDate(el);
+    }
+
+    if (el.is("#email_textbox")){
+        isValid = validateEmail(el);
+    }
+
+    if (el.is(".name")) {
+        isValid = validateName(el);
```

Comment:
For alias fields, you should validate name for all fields not just current field.

Reply:
I'm testing any field that has a `class="name"` in it, and I had to add a `<span>` around each input field so the `has-error` class had something to attach to. The first time I tried this, it seemed to break the functionality of the aliases, but now it seems ok - not sure what I did differently. Anyhow, the change is in `alias_field.html`.